system/src/system_task.cpp

```diff
@@ -368,12 +368,13 @@ void system_delay_ms_non_threaded(unsigned long ms, bool force_no_background_loo
 
 void system_delay_ms(unsigned long ms, bool force_no_background_loop=false)
 {
-    if (system_thread_get_state(NULL)!=0) {
-        HAL_Delay_Milliseconds(ms);
+    if (system_thread_get_state(NULL) == spark::feature::DISABLED &&
```

Comment:
Ok, I see. I wasn't considering the case of creating additional threads since this release is focused on just the system/application thread.